### PR TITLE
Link update options

### PR DESCRIPTION
**Pull request by manuelpuyol**
Link will now accept `variant`, `underline` and also `tag: :span`.
Updates made according to https://primer-css-git-mkt-color-modes-docs-primer.vercel.app/css/components/links

image

### CODE REVIEW

File: app/components/primer/link_component.rb [hunk 1]
```diff
@@ -5,27 +5,57 @@ module Primer
...
+        "no-underline" => !underline
       )
     end
 
     def call
       render(Primer::BaseComponent.new(**@system_arguments)) { content }
     end
+
+    def before_render
+      raise ArgumentError, "href is required when using <a> tag" if @system_arguments[:tag] == :a && @system_arguments[:href].nil? && !Rails.env.production?
```
Comment (srt32) [inline app/components/primer/link_component.rb:58]: I wonder if we could call this out more explicitly in the docs string above?

File: demo/app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -4,7 +4,7 @@
     <title>Demo</title>
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
-    <link href="https://unpkg.com/@primer/css-next/dist/primer.css" rel="stylesheet" />
+    <link href="https://unpkg.com/@primer/css-next@canary/dist/primer.css" rel="stylesheet" />
```
Comment (srt32) [inline demo/app/views/layouts/application.html.erb:7]: Could you help me understand why we want this change?

File: app/components/primer/link_component.rb (hunk 1 above)
Comment (srt32) [reply app/components/primer/link_component.rb:58]: It's not clear to me we'd raise and then raise only in non prod environments. Maybe this?

> # @param href [String] URL to be used for the Link. Required if tag is :a. If the requirements are not met an error will be raised in non production environments. In production, an empty link element will be rendered.

**Review by srt32 — APPROVED** [review]

File: app/components/primer/link_component.rb [hunk 3]
```diff
@@ -5,27 +5,57 @@ module Primer
...
+    # @example Without underline
+    #   <%= render(Primer::LinkComponent.new(href: "#", underline: false)) { "Link" } %>
     #
-    # @param href [String] URL to be used for the Link
-    # @param muted [Boolean] Uses light gray for Link color, and blue on hover
+    # @example Span as link
+    #   <%= render(Primer::LinkComponent.new(tag: :span)) { "Span as a link" } %>
+    #
+    # @param tag [String]  <%= one_of(Primer::LinkComponent::TAG_OPTIONS) %>
+    # @param href [String] URL to be used for the Link. Required if tag is `:a`.
```
Comment (manuelpuyol) [inline app/components/primer/link_component.rb:35]:
```suggestion
    # @param href [String] URL to be used for the Link. Required if tag is `:a`. If the requirements are not met an error will be raised in non production environments. In production, an empty link element will be rendered.
```

**Pull request merged**